core/src/avm2/globals/flash/net/XMLSocket.as

```diff
@@ -1,23 +1,84 @@
-package flash.net {
-    import flash.events.EventDispatcher;
-    import __ruffle__.stub_constructor;
-    import __ruffle__.stub_method;
-
-    public class XMLSocket extends EventDispatcher {
-        public function XMLSocket(host: String = null, port: int = 0) {
-            stub_constructor("flash.net.XMLSocket");
+package flash.net
+{
+    import flash.net.Socket;
+    import flash.events.ProgressEvent;
+    import flash.events.DataEvent;
+    import flash.utils.ByteArray;
+
+    // NOTE: Extending from Socket is a implementation detail. It is not possible for SWFs to
+    // use Socket methods as Flash's playerglobal doesn't include those methods, unless
+    // SWF author intentionally tried, which wouldn't make much sense.
+    public class XMLSocket extends Socket
```

Comment:
We try to keep things as close to Flash as possible, as this is detectable (and therefor potentially bug-inducing) via as3.

I'd suggest keeping a `private var socket: Socket` instead, and wrapping that.

Reply:
Done.